tukaan/graphics/canvas.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from tukaan._base import TkWidget, WidgetBase
+from tukaan._tcl import Tcl
+from .equipment import Brush, Pen
+from .gradient import LinearGradient, RadialGradient, Gradient
+from tukaan.colors import Color
+
+
+class Canvas(WidgetBase):
+    _tcl_class = "::tkp::canvas"
+
+    def __init__(
+        self,
+        parent: TkWidget,
+        *,
+        width: int | None = 400,
+        height: int | None = 400,
+        bg_color: Color | str | None = None,
+    ) -> None:
+        super().__init__(
+            parent,
+            width=width,
+            height=height,
+            background=bg_color,
+            borderwidth=0,
+            highlightthickness=0,
+        )
+
+        self.brush = Brush()
+        self.pen = Pen()
+
+        self._gradient_superclass = Gradient(self)
+        self.linear_gradient = LinearGradient(self)
+        self.radial_gradient = RadialGradient(self)
```

Comment:
This is super unintuitive, and makes no sense.

Unfortunately we can't make it totally independent from the camvas wodget, because that's not how TkPath works, but it shouldn't be a "method".

Plus another thing came into my mind, that's actually a serious problem. There are random helper objects that are mutable and there are others that are immutable. This is because the objects that are mutable are actual Tk stuff, and others are just simple Python objects, e.g dataclasses, which can't notify the widget is an attribute is changed. This is NOT good. Something must be done.

Mutable:
* `Font`
* gradients
* Control variables

Immutable
* `Color`
* `Brush` and `Pen`
* `ArrowHead`